tests/zfs-tests/tests/functional/cli_root/zdb/zdb_objset_id.ksh

```diff
@@ -89,13 +88,9 @@ do
 		log_fail "zdb -NNNNNN $TESTPOOL/$id $obj failed (file1 not in zdb output)"
 done
 
-if is_linux; then
-	output=$(ls -1 /proc/spl/kstat/zfs/$TESTPOOL | grep objset- | tail -1)
-	objset_hex=${output#*-}
-	name_from_proc=$(grep dataset_name /proc/spl/kstat/zfs/$TESTPOOL/$output | cut -d' ' -f3)
-	log_note "checking zdb output for $name_from_proc"
-	log_must eval "zdb -dddddd $TESTPOOL/$objset_hex | grep -q \"$name_from_proc\""
-fi
+name_from_proc=$(kstat objset-$objset_hex.dataset_name $TESTPOOL)
```

Comment:
Above for FreeBSD in pool presence you use "kstat.zfs.$pool.misc.$stat" format.  But the sysctl looks like "kstat.zfs.hdd.dataset.objset-0x36.dataset_name=hdd", i.e. with "dataset" instead of "misc" after the pool name.  But there are couple items "txgs" and "reads" without any prefix.  Linux as I see does not have that "dataset"/"misc" layer.  What a mess...

Reply:
Oh wow, you're right (also, how did I miss that).

I'm not sure what to do other than add special-case checks in `kstat()`, or, perhaps, `kstat_dataset()` (and perhaps `kstat_pool()` to go with it).

(Longer term I'd want them to be uniform across platforms, and I've made some small starts on it, but I don't want to rename anything until we're sure we don't have to change them again.)

Reply:
v2 pushed, which takes care of all of this and probably adds another bunch of complication!